seed/seedwriter/writer.go

```diff
@@ -462,6 +579,11 @@ func (w *Writer) Downloaded() (complete bool, err error) {
 	for _, sn := range w.snapsFromModel {
 		info := sn.Info
 		if !sn.local {
+			if info.ID() == "" {
```

Comment:
`if !sn.local && info.ID() == "" {...` ?

Reply:
the first if is checking context the 2nd is a sanity check, so I prefer them split